src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => delete state[key]);
+        break;
+
+      case 'clear':
+        Object.keys(state).forEach((key) => delete state[key]);
+        break;
+
+      default:
+        window.alert('Unknown action type: ' + action.type);
```

Comment:
It would be better to `throw Error` here, as it allows us to handle it properly 😉